sdkms/apps_generated.go

```diff
@@ -174,205 +247,321 @@ func (x *AppOauthConfig) UnmarshalJSON(data []byte) error {
-	CertNotAfter *Time           `json:"cert_not_after,omitempty"`
-	CreatedAt    Time            `json:"created_at"`
-	Creator      Principal       `json:"creator"`
-	DefaultGroup *UUID           `json:"default_group,omitempty"`
-	Description  *string         `json:"description,omitempty"`
-	Enabled      bool            `json:"enabled"`
-	Groups       AppGroups       `json:"groups"`
-	Interface    *string         `json:"interface,omitempty"`
-	LastusedAt   *Time           `json:"lastused_at,omitempty"`
-	Name         string          `json:"name"`
-	OauthConfig  *AppOauthConfig `json:"oauth_config,omitempty"`
+type LastAppOperationTimestamp struct {
+	Generic           *uint64 `json:"generic,omitempty"`
+	Tokenization      *uint64 `json:"tokenization,omitempty"`
+	Tep               *uint64 `json:"tep,omitempty"`
+	Accelerator       *uint64 `json:"accelerator,omitempty"`
+	SecretsManagement *uint64 `json:"secrets_management,omitempty"`
 }
 
-type AppRequest struct {
-	AddGroups    *AppGroups      `json:"add_groups,omitempty"`
-	AppType      *string         `json:"app_type,omitempty"`
-	Credential   *AppCredential  `json:"credential,omitempty"`
-	DefaultGroup *UUID           `json:"default_group,omitempty"`
-	DelGroups    *[]UUID         `json:"del_groups,omitempty"`
-	Description  *string         `json:"description,omitempty"`
-	Enabled      *bool           `json:"enabled,omitempty"`
-	Interface    *string         `json:"interface,omitempty"`
-	ModGroups    *AppGroups      `json:"mod_groups,omitempty"`
-	Name         *string         `json:"name,omitempty"`
-	OauthConfig  *AppOauthConfig `json:"oauth_config,omitempty"`
-	SecretSize   *uint32         `json:"secret_size,omitempty"`
+// Query params for Get all apps API
+type ListAppsParams struct {
+	// Group for which the associated apps should be retrived.
+	GroupID *UUID `json:"group_id,omitempty"`
+	// Maximum number of apps to return. Default limit is 1001.
+	Limit *uint `json:"limit,omitempty"`
+	// Number of apps to skip from the beginning/start.
+	Offset *uint `json:"offset,omitempty"`
+	// Sort apps by app_id in ascending or descending order.
+	Sort AppSort `json:"sort"`
+	// Flag specifying if group permissions should be returned with the apps.
+	GroupPermissions *bool `json:"group_permissions,omitempty"`
+	// Specify role of the apps.
+	Role *AppRole `json:"role,omitempty"`
 }
 
-type AppResetSecretRequest struct {
-	// Size of app's secret in bytes.
-	SecretSize *uint32 `json:"secret_size,omitempty"`
+func (x ListAppsParams) urlEncode(v map[string][]string) error {
+	if x.GroupID != nil {
+		v["group_id"] = []string{fmt.Sprintf("%v", *x.GroupID)}
+	}
+	if x.Limit != nil {
+		v["limit"] = []string{fmt.Sprintf("%v", *x.Limit)}
+	}
+	if x.Offset != nil {
+		v["offset"] = []string{fmt.Sprintf("%v", *x.Offset)}
+	}
+	if err := x.Sort.urlEncode(v); err != nil {
+		return err
+	}
+	if x.GroupPermissions != nil {
+		v["group_permissions"] = []string{fmt.Sprintf("%v", x.GroupPermissions)}
```

Comment:
This line is wrong, it will print the address of boolean, instead of true/false
```suggestion
		v["group_permissions"] = []string{fmt.Sprintf("%v", *x.GroupPermissions)}
```